### ISSUE TITLE
Compiling 2.3.1, getting "Segmentation fault (core dumped)"

### ISSUE THREAD

**Issue by arpieb** [certifi/erlang-certifi#33]
I'm trying to rebuild an Elixir project that was rebuilding fine a few weeks ago, and now segfaults on compilation of this module:

```
$ mix deps.compile --all
===> Compiling parse_trans
===> Compiling mimerl
===> Compiling metrics
===> Compiling unicode_util_compat
===> Compiling idna
==> ssl_verify_fun (compile)
===> Compiling certifi
Segmentation fault (core dumped)
** (Mix) Could not compile dependency :certifi, "/home/rbates/.mix/rebar3 bare compile --paths "/home/rbates/src/gdelt_pull/_build/dev/lib/*/ebin"" command failed. You can recompile this dependency with "mix deps.compile certifi", update it with "mix deps.update certifi" or clean it with "mix deps.clean certifi"
```

I have cleaned, updated, recompiled, etc to no effect.  My current system config is:

- Erlang/OTP 21 [erts-10.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]
- Mix 1.7.0 (compiled with Erlang/OTP 20)
- Elixir 1.7.0 (compiled with Erlang/OTP 20)
- Ubuntu 16.04.5 LTS
- Linux 4.15.0-30-generic #32~16.04.1-Ubuntu SMP Thu Jul 26 20:25:39 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

Other dependencies in the project are:

-   certifi 2.3.1
-   hackney 1.13.0
-   httpoison 1.2.0
-   idna 5.1.2
-   metrics 1.0.1
-   mimerl 1.0.2
-   parse_trans 3.2.0
-   ssl_verify_fun 1.1.1
-   unicode_util_compat 0.3.1

Please let me know if you need any additional information.  I'll keep trying to suss it out and will post any updates from my end as well.

Thanks!

**Comment by benoitc** [comment]
Can you describe how you build it and the RAM you have in the machine or vm?

**Comment by arpieb** [comment]
The build machine has 64G RAM, 6TB drivespace, and an i7-7800X CPU @ 3.50GHz so it's not resource constrained in any way.  Previous versions of Elixir and BEAM have compiled and run just fine on this machine as well as a Lubuntu system that I sometimes use this code on.

Interestingly enough, I rebooted the machine last night (updated the GPU drivers) and now it's compiling fine.  I just tried the following build options (which were breaking before the reboot) and they all compiled with no errors:

- `mix clean; mix deps.compile`
- `mix clean; mix escript.build`
- `mix clean; mix compile`
- `mix clean; iex -S mix`

Must have had something in limbo post-update.  Closing!  Thanks for the quick response!
